Source/Particles/Collision/BackgroundStopping/BackgroundStopping.cpp

```diff
@@ -0,0 +1,269 @@
+        m_background_type = BackgroundStoppingType::IONS;
+    } else {
+        AMREX_ALWAYS_ASSERT_WITH_MESSAGE(false, "background_type must be either electrons or ions");
+    }
+
+    amrex::Real background_density;
+    std::string background_density_str;
+    if (queryWithParser(pp_collision_name, "background_density", background_density)) {
+        AMREX_ALWAYS_ASSERT_WITH_MESSAGE(background_density > 0,
+                 "For background stopping, the background density must be greater than 0");
+        m_background_density_parser = makeParser(std::to_string(background_density), {"x", "y", "z", "t"});
+    } else if (pp_collision_name.query("background_density(x,y,z,t)", background_density_str)) {
+        m_background_density_parser = makeParser(background_density_str, {"x", "y", "z", "t"});
+    } else {
+        AMREX_ALWAYS_ASSERT_WITH_MESSAGE(false,
+                 "For background stopping, the background density must be specified.");
+    }
+
+    amrex::Real background_temperature;
+    std::string background_temperature_str;
+    if (queryWithParser(pp_collision_name, "background_temperature", background_temperature)) {
+        AMREX_ALWAYS_ASSERT_WITH_MESSAGE(background_temperature > 0,
+                 "For background stopping, the background temperature must be greater than 0");
+        m_background_temperature_parser = makeParser(std::to_string(background_temperature), {"x", "y", "z", "t"});
+    } else if (pp_collision_name.query("background_temperature(x,y,z,t)", background_temperature_str)) {
+        m_background_temperature_parser = makeParser(background_temperature_str, {"x", "y", "z", "t"});
+    } else {
+        AMREX_ALWAYS_ASSERT_WITH_MESSAGE(false,
+                 "For background stopping, the background temperature must be specified.");
+    }
+
+    m_background_density_func = m_background_density_parser.compile<4>();
+    m_background_temperature_func = m_background_temperature_parser.compile<4>();
+
+    if (m_background_type == BackgroundStoppingType::ELECTRONS) {
+        m_background_mass = PhysConst::m_e;
+        queryWithParser(pp_collision_name, "background_mass", m_background_mass);
+    } else if (m_background_type == BackgroundStoppingType::IONS) {
+        getWithParser(pp_collision_name, "background_mass", m_background_mass);
+        getWithParser(pp_collision_name, "background_charge_state", m_background_charge_state);
+    }
+    AMREX_ALWAYS_ASSERT_WITH_MESSAGE(m_background_mass > 0,
+             "For background stopping, the background mass must be greater than 0");
+
+}
+
+void
+BackgroundStopping::doCollisions (amrex::Real cur_time, MultiParticleContainer* mypc)
+{
+    WARPX_PROFILE("BackgroundStopping::doCollisions()");
+    using namespace amrex::literals;
+
+    // Note that the lowest level time step is used for all levels since the
+    // collision operation will be done only once per step at the lowest level.
+    const amrex::Real dt = WarpX::GetInstance().getdt(0);
```

Comment:
Should we actually multiply this quantity by `m_ndt` because we are performing collisions on every `m_ndt` timestep?
```suggestion
    const amrex::Real dt = WarpX::GetInstance().getdt(0) * m_ndt;
```
Could you actually use the `ndt` feature in the automated test (e.g. `ndt = 2` or `ndt = 3`, just to check that it gives the correct results?

Reply:
I'll add your suggestion. Since all particles are affected by this and since this should be a fast calculation and not take much time, I would guess that there is little reason to use the `ndt` option.

Reply:
A related comment, in all of the different kinds of collisions, the same multiplication by `m_ndt` is done and the check of `int(std::floor(cur_time/dt)) % m_ndt != 0`. Perhaps these two lines could be moved up a level into the collision handler routine that calls `doCollisions`? Also, like the current time, the `dt` should be passed in from the call in `Evolve`. I'll make a separate PR with this clean up.